packages/chip/src/Chip/Chip.spec.tsx

```diff
@@ -0,0 +1,328 @@
+      });
+
+      test('renders correct label when chipCharacterLimit is 10 and chipTruncationLocation is start', () => {
+        const { chipText } = renderChip({
+          label: longLabel,
+          chipCharacterLimit: 10,
+          chipTruncationLocation: 'start',
+        });
+        expect(chipText).toHaveTextContent('…h crush');
+      });
+
+      test('renders correct label when chipCharacterLimit is 10 and chipTruncationLocation is middle', () => {
+        const { chipText } = renderChip({
+          label: longLabel,
+          chipCharacterLimit: 10,
+          chipTruncationLocation: 'middle',
+        });
+        expect(chipText).toHaveTextContent('cru…rush');
+      });
+    });
+  });
+
+  describe('dismiss button', () => {
+    test('renders when onDismiss is passed', () => {
+      const onDismiss = jest.fn();
+      const { queryByTestId } = renderChip({ onDismiss });
+      const button = queryByTestId('chip-dismiss-button');
+      expect(button).toBeInTheDocument();
+    });
+
+    test('fires onDismiss callback', () => {
+      const onDismiss = jest.fn();
+      const { queryByTestId } = renderChip({ onDismiss });
+      const button = queryByTestId('chip-dismiss-button');
+      userEvent.click(button!);
+      expect(onDismiss).toHaveBeenCalledTimes(1);
+    });
+
+    test('renders with default aria-label', () => {
+      const onDismiss = jest.fn();
+      const { queryByTestId } = renderChip({ onDismiss });
+      const button = queryByTestId('chip-dismiss-button');
+      expect(button?.getAttribute('aria-label')).toBe('Deselect chip');
+    });
+
+    test('renders with updated aria-label when dismissButtonAriaLabel is passed', () => {
+      const onDismiss = jest.fn();
+      const { queryByTestId } = renderChip({
+        onDismiss,
+        dismissButtonAriaLabel: 'new aria label',
+      });
+      const button = queryByTestId('chip-dismiss-button');
+      expect(button?.getAttribute('aria-label')).toBe('new aria label');
+    });
+
+    test('renders aria-disabled attribute when disabled is set', () => {
+      const onDismiss = jest.fn();
+      const { queryByTestId } = renderChip({ disabled: true, onDismiss });
+      const button = queryByTestId('chip-dismiss-button');
+      expect(button?.getAttribute('aria-disabled')).toBe('true');
+    });
+
+    test('does nothing when disabled button is clicked', () => {
+      const onDismiss = jest.fn();
+      const { queryByTestId } = renderChip({ disabled: true, onDismiss });
+      const button = queryByTestId('chip-dismiss-button');
+      expect(() => userEvent.click(button!)).toThrow();
+    });
+  });
+
+  describe('dismiss button with truncated text', () => {
+    test('renders when onDismiss is passed', () => {
+      const onDismiss = jest.fn();
+      const { queryByTestId } = renderChip({ onDismiss, label: longLabel });
+      const button = queryByTestId('chip-dismiss-button');
+      expect(button).toBeInTheDocument();
+    });
```

Comment:
nit - the description in `describe` doesn't accurately describe this test case

Reply:
thanks, looks like im missing the truncated limit and location